sycl/source/detail/config.hpp

```diff
@@ -163,6 +165,118 @@ template <> class SYCLConfig<SYCL_PI_TRACE> {
   }
 };
 
+template <> class SYCLConfig<SYCL_DEVICE_TRIPLE> {
+  using BaseT = SYCLConfigBase<SYCL_DEVICE_TRIPLE>;
+
+public:
+  static std::vector<triple> *get() {
+    static bool Initialized = false;
+    static std::string String;
+    static std::vector<triple> TripleList;
+
+    // Configuration parameters are processed only once, like reading a string
+    // from environment and converting it into a typed object.
+    if (Initialized) {
+      if (TripleList.size()) {
+        return &TripleList;
+      } else {
+        return nullptr;
+      }
+    }
+
+    const std::array<std::pair<std::string, info::device_type>, 5>
+        SyclDeviceTypeMap = {{{"host", info::device_type::host},
+                              {"cpu", info::device_type::cpu},
+                              {"gpu", info::device_type::gpu},
+                              {"acc", info::device_type::accelerator},
+                              {"*", info::device_type::all}}};
+    const std::array<std::pair<std::string, backend>, 4> SyclBeMap = {
+        {{"opencl", backend::opencl},
+         {"level0", backend::level_zero},
+         {"level_zero", backend::level_zero},
+         {"cuda", backend::cuda}}};
+
+    Initialized = true;
+    const char *ValStr = BaseT::getRawValue();
+    if (ValStr) {
+      String = ValStr;
+      std::transform(String.begin(), String.end(), String.begin(), ::tolower);
+      bool MoreTriple = true;
+      size_t Pos = 0;
+      while (MoreTriple) {
```

Comment:
I think SYCLConfig is the wrong place for parsing the actual value. SYCLConfig just provides ability to get some control from env var, config file or define, and cache it. I suggest adding(and use) triple c'tor which takes a triple string and parse it.

Reply:
@romanovvlad , My original implementation was parsing the string as you mentioned.
But, @alexbatashev told me to use SYCLConfig.
I can do either way, but please choose only one (eihter string parse or SYCLConfig).

Reply:
@bso-intel I guess, @romanovvlad means that you need to obtain a string through SYCLConfig, but parse it somewhere else. SYCLConfig allows you not only get values from env vars, but also from config files and compile macros.

Reply:
Thanks, @alexbatashev.
If we cache the raw string instead of sycl::Triple obejct, we have to parse the cached string over and over again whenever SYCLConfig::get() is called, right?
Unless we need to parse the string differently, I think it makes more sense to cache the TripleList instead of the raw string.

Reply:
The raw string is cached anyway. I'm just saying that, I think, SYCLConfig is not right place for doing all this string manipulations. I suggest moving this logic to triple constructors.

Reply:
@romanovvlad , yes exactly.  As we discussed today. :)